proxy/start_container_interceptor.go

```diff
@@ -24,10 +26,25 @@ func (i *startContainerInterceptor) InterceptResponse(r *http.Response) error {
 	}
 
 	cidrs, ok := weaveCIDRsFromConfig(container.Config)
-	if !ok {
+	if !ok && !i.forceIPAM {
 		Debug.Print("No Weave CIDR, ignoring")
 		return nil
 	}
+	if len(cidrs) == 0 {
+		api, err := weaveapi.Find(i.client)
+		if err != nil {
+			Warning.Printf("Failed to connect to weave container: %s", err)
+			return err
+		}
+
+		ipamCIDR, err := api.AllocateIPFor(container.ID)
+		if err != nil {
+			Warning.Printf("Determining container %s IP via IPAM failed: %v", container.ID, err)
+			return err
+		}
+
+		cidrs = strings.Fields(ipamCIDR)
+	}
 	Info.Printf("Attaching container %s with WEAVE_CIDR \"%s\" to weave network", container.ID, strings.Join(cidrs, " "))
```

Comment:
Though, as noted above, the whole 'talk to weave' thing is wrong. Ditch that and the message is correct again, i.e. in the IPAM case it would show `with WEAVE_CIDR ""`, which is ok.